distributed/worker.py

```diff
@@ -1792,31 +1767,32 @@ def ensure_communicating(self):
                     changed = True
                     continue
 
-                if self.task_state.get(key) != "waiting":
+                ts = self.tasks[key]
+                if ts.state != "waiting":
                     self.log.append((key, "communication pass"))
                     self.data_needed.popleft()
                     changed = True
                     continue
 
-                deps = self.dependencies[key]
+                deps = ts.dependencies
                 if self.validate:
-                    assert all(dep in self.dep_state for dep in deps)
+                    assert all(hasattr(dep, "state") for dep in deps)
```

Comment:
I would think that TaskState objects always have a `.state` attribute

Reply:
yeah, I think this was from an earlier debugging pass, cleaning it up now.